Issue when I run ng lint in a project which is minimal

Step 1- Create a project using following command:
```
ng new test-app --enable-ivy=true --minimal=true --prefix=test --routing=true --skipInstall=true --skipTests=true --style=scss
```
Step 2- Run following command
```
ng lint
```

**First issue:** You'll receive the following error:

```
error TS5058: The specified path does not exist: 'test-app\tsconfig.spec.json'.
```

There's no tsconfig.spec.json because --minimal parameter of the first command.
In angular.json, there's a lint section which accepts two tsconfig files instead of one!

**The second issue:**

There is no tslint.json at all!

If you provide it (by copying it from another angular project), you'll see **3rd issue**:

```Could not find custom rule directory: codelyzer```

You've to install it using

```
npm i codelyzer --save-dev
```

You can enjoy using ng lint with a minimally created project.

But it seems that @angular/cli needs some changes related to these issues.

Thanks in advance.

--- a/packages/schematics/angular/application/index.ts
+++ b/packages/schematics/angular/application/index.ts
@@ -261,7 +261,7 @@ function addAppToWorkspaceFile(options: ApplicationOptions, appDir: string): Rul
           scripts: [],
         },
       },
-      lint: {
+      lint: options.minimal ? undefined : {
         builder: Builders.TsLint,
         options: {
           tsConfig: [

--- a/packages/schematics/angular/application/index_spec.ts
+++ b/packages/schematics/angular/application/index_spec.ts
@@ -174,13 +174,14 @@ describe('Application Schematic', () => {
     expect(karmaConf).toContain(`dir: require('path').join(__dirname, '../../coverage/foo')`);
   });
 
-  it('minimal=true should not create e2e and test targets', async () => {
+  it('minimal=true should not create e2e, lint and test targets', async () => {
     const options = { ...defaultOptions, minimal: true };
     const tree = await schematicRunner.runSchematicAsync('application', options, workspaceTree)
       .toPromise();
     const config = JSON.parse(tree.readContent('/angular.json'));
     const architect = config.projects.foo.architect;
     expect(architect.test).not.toBeDefined();
+    expect(architect.e2e).not.toBeDefined();
     expect(architect.e2e).not.toBeDefined();
   });
 
@@ -270,6 +271,18 @@ describe('Application Schematic', () => {
 
       const packageJson = JSON.parse(tree.readContent('package.json'));
       expect(packageJson.devDependencies['@angular-devkit/build-angular']).toBeUndefined();
+    });
+
+    it('should set the lint tsConfig option', async () => {
+      const tree = await schematicRunner.runSchematicAsync('application', defaultOptions, workspaceTree)
+        .toPromise();
+      const workspace = JSON.parse(tree.readContent('/angular.json'));
+      const lintOptions = workspace.projects.foo.architect.lint.options;
+      expect(lintOptions.tsConfig).toEqual([
+        'projects/foo/tsconfig.app.json',
+        'projects/foo/tsconfig.spec.json',
+        'projects/foo/e2e/tsconfig.json',
+      ]);
     });
   });
 

--- a/packages/schematics/angular/e2e/index_spec.ts
+++ b/packages/schematics/angular/e2e/index_spec.ts
@@ -98,17 +98,5 @@ describe('Application Schematic', () => {
       expect(e2eOptions.protractorConfig).toEqual('projects/foo/e2e/protractor.conf.js');
       expect(e2eOptions.devServerTarget).toEqual('foo:serve');
     });
-
-    it('should set the lint options', async () => {
-      const tree = await schematicRunner.runSchematicAsync('e2e', defaultOptions, applicationTree)
-        .toPromise();
-      const workspace = JSON.parse(tree.readContent('/angular.json'));
-      const lintOptions = workspace.projects.foo.architect.lint.options;
-      expect(lintOptions.tsConfig).toEqual([
-        'projects/foo/tsconfig.app.json',
-        'projects/foo/tsconfig.spec.json',
-        'projects/foo/e2e/tsconfig.json',
-      ]);
-    });
   });
 });

